src/components/modals/baseModal.vue

```diff
@@ -0,0 +1,37 @@
+<template>
+  <Teleport to="#modals"
+    ><UseFocusTrap :options="options">
+      <div
+        class="fixed inset-0 bg-gray-900 bg-opacity-40"
+        ref="target"
```

Comment:
you're using the UseFocusTrap component so the ref isn't used and needed.
So, or you only use the component (https://vueuse.org/integrations/useFocusTrap/#using-component) or you only use the composable (https://vueuse.org/integrations/useFocusTrap/#usage).